Zend/zend_operators.c

```diff
@@ -2737,11 +2737,21 @@ ZEND_API zend_result ZEND_FASTCALL decrement_function(zval *op1) /* {{{ */
 					zval_ptr_dtor_str(op1);
 					ZVAL_DOUBLE(op1, dval - 1);
 					break;
-				default:
+				default: {
+					/* Error handler can unset the variable */
+					zend_string *zstr = Z_STR_P(op1);
+					GC_TRY_ADDREF(zstr);
 					zend_error(E_DEPRECATED, "Decrement on non-numeric string has no effect and is deprecated");
 					if (EG(exception)) {
+						GC_TRY_DELREF(zstr);
```

Comment:
`zend_string_release`

Reply:
Done, and also amended the previous usage